packages/serverpod/lib/src/endpoints/insights.dart

```diff
@@ -215,32 +216,33 @@ class InsightsEndpoint extends Endpoint {
   }
 
   /// Returns the target and live database definitions. See
-  /// [getTargetDatabaseDefinition] and [getLiveDatabaseDefinition] for more
+  /// [getTargetTableDefinition] and [getLiveDatabaseDefinition] for more
   /// details.
   Future<DatabaseDefinitions> getDatabaseDefinitions(Session session) async {
-    var target = await getTargetDatabaseDefinition(session);
+    var targetTables = await getTargetTableDefinition(session);
     var live = await getLiveDatabaseDefinition(session);
     var installedMigrations =
         await DatabaseAnalyzer.getInstalledMigrationVersions(session);
 
-    var modules = MigrationVersions.listAvailableModules();
+    var versions = MigrationVersions.listVersions();
 
     var latestAvailableMigrations = <DatabaseMigrationVersion>[];
-
-    for (var module in modules) {
-      var version =
-          Serverpod.instance.migrationManager.getLatestVersion(module);
-      latestAvailableMigrations.add(
-        DatabaseMigrationVersion(
-          module: module,
-          version: version,
-        ),
+    if (versions.isNotEmpty) {
+      var version = versions.last;
+      var file = MigrationConstants.databaseDefinitionJSONPath(
+        Directory.current,
+        version,
       );
+      var data = await file.readAsString();
+      var databaseDefinition = session.serverpod.serializationManager
+          .decode<DatabaseDefinition>(data);
+
+      latestAvailableMigrations = databaseDefinition.installedModules;
```

Comment:
Question - Shouldn't this be the list of available migrations on disk? If I understand this code correctly we now return the installed modules in tha latest migration.

Reply:
We cannot get them as they are in the pub spec dependencies. This is the next best thing we are able to do. Here we will catch any generated migration that is not applied.

Reply:
I think this problem can probably be solved inside insights instead, by checking the dependency.